osu.Game/Overlays/Wiki/WikiSidebar.cs

```diff
@@ -0,0 +1,116 @@
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Containers.Markdown;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+
+namespace osu.Game.Overlays.Wiki
+{
+    public class WikiSidebar : OverlaySidebar
+    {
+        private FillFlowContainer tableOfContents;
+
+        protected override Drawable CreateContent() => new FillFlowContainer
+        {
+            Direction = FillDirection.Vertical,
+            RelativeSizeAxes = Axes.X,
+            AutoSizeAxes = Axes.Y,
+            Children = new Drawable[]
+            {
+                new OsuSpriteText
+                {
+                    Text = "CONTENTS",
+                    Font = OsuFont.GetFont(size: 12, weight: FontWeight.Bold),
+                },
+                tableOfContents = new FillFlowContainer
+                {
+                    Direction = FillDirection.Vertical,
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                }
+            },
+        };
+
+        public void AddToc(HeadingBlock headingBlock, MarkdownHeading heading)
+        {
+            switch (headingBlock.Level)
+            {
+                case 2:
+                case 3:
+                    string title = getTitle(headingBlock.Inline);
+                    tableOfContents.Add(new TocTitle(title, heading, headingBlock.Level == 3));
+                    break;
+            }
+        }
+
+        private string getTitle(ContainerInline containerInline)
+        {
+            foreach (var inline in containerInline)
+            {
+                switch (inline)
+                {
+                    case LiteralInline literalInline:
+                        return literalInline.Content.ToString();
+
+                    case LinkInline { IsImage: false } linkInline:
+                        return getTitle(linkInline);
+                }
+            }
+
+            return string.Empty;
+        }
+
+        private class TocTitle : OsuHoverContainer
+        {
+            [Resolved]
+            private OverlayScrollContainer scrollContainer { get; set; }
+
+            private readonly MarkdownHeading target;
+
+            private readonly OsuTextFlowContainer textFlow;
+
+            public TocTitle(string text, MarkdownHeading target, bool subtitle = false)
+            {
+                this.target = target;
+
+                RelativeSizeAxes = Axes.X;
+                AutoSizeAxes = Axes.Y;
+                Child = textFlow = new OsuTextFlowContainer(t =>
+                {
+                    t.Font = OsuFont.GetFont(size: subtitle ? 12 : 15);
+                }).With(f =>
+                {
+                    f.AddText(text);
+                    f.RelativeSizeAxes = Axes.X;
+                    f.AutoSizeAxes = Axes.Y;
+                });
+                Margin = new MarginPadding { Top = subtitle ? 5 : 10 };
```

Comment:
Don't think this is entirely right. As far as I can see from web, the spacing is:

* 2px bottom margin on `.wiki-toc-list__link`
* 5px bottom margin on `.wiki-toc-list--top > .wiki-toc-list__item`
* 10px bottom margin on `.wiki-toc-list`

In other words:

* All links (both top-level and sub-links) have a 2px bottom margin
* Singular top-level links without child entries have a 5px bottom margin
* Top-level links with child entries have a 10px bottom margin

Reply:
I refactor this and create `WikiTableOfContents`.